independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java

```diff
@@ -592,23 +600,20 @@ public List<ClassPathElement> getElementsWithResource(String name, boolean local
         if (parent instanceof QuarkusClassLoader && !localOnly) {
             ret.addAll(((QuarkusClassLoader) parent).getElementsWithResource(name));
         }
-        ClassPathElement[] classPathElements = getState().loadableResources.get(name);
-        if (classPathElements == null) {
+        List<ClassPathElement> classPathElements = getClassPathResourceIndex().getClassPathElements(name);
+        if (classPathElements.isEmpty()) {
             return ret;
         }
-        ret.addAll(Arrays.asList(classPathElements));
+        ret.addAll(classPathElements);
         return ret;
     }
 
-    public List<String> getLocalClassNames() {
+    public Set<String> getLocalClassNames() {
```

Comment:
Why this change of type?

Reply:
Because that's the semantic we have. A big random bag of unique strings :).

Reply:
Yeah, but what do we gain by changing it to a `Set`?

Reply:
The values are coming from Sets now so I decided to be consistent rather than going from a Set to a List, especially since that's the semantic we want.